public/docs/ts/latest/guide/router.jade

```diff
@@ -884,13 +889,24 @@ a#hero-detail-ctor
 :marked
   Later, in the `ngOnInit` method,
   we use the `ActivatedRoute` service to retrieve the parameters for our route.
-  Since our parameters are provided as an `Observable`, we use the _forEach_ method to retrieve them for the `id` parameter by name and
+  Since our parameters are provided as an `Observable`, we use the _switchMap_ operator to provide them for the `id` parameter by name and
   tell the `HeroService` to fetch the hero with that `id`.
 
 +makeExcerpt('app/heroes/hero-detail.component.ts (ngOnInit)', 'ngOnInit')
 
 .l-sub-section
   :marked
+    The `switchMap` operator allows you to perform an action with the current value of the `Observable`,
+    and map it to a new `Observable`. As with many `rxjs` operators, `switchMap` handles
+    an `Observable` as well as a `Promise` to retrieve the value they emit.
+
+    We could have used the `flatMap` operator, which works the same way with one exception.
+    The `switchMap` operator will also cancel any in-flight requests if our user re-navigates to the route
```

Comment:
Upon further reflection, while I remain profoundly uncomfortable with what seems like a distraction to me, I'm letting it slide.